feature_engine/datetime/datetime.py

```diff
@@ -250,13 +275,36 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Check if input data contains same number of columns as dataframe used to fit.
         _check_input_matches_training_df(X, self.n_features_in_)
 
+        # special case index
+        if self.variables_ == ["index"]:
```

Comment:
We could use either self.variables_ is None or self.variables =="index" without the underscore.